cli/daemon/run/testdata/echo_client/client.ts

```diff
@@ -66,7 +66,7 @@ export interface ClientOptions {
      * request either by passing in a static object or by passing in
      * a function which returns a new object for each request.
      */
-    auth?: echo.AuthParams | AuthDataGenerator
+    auth?: (echo.AuthParams | null) | AuthDataGenerator
```

Comment:
This is a really complex type signature, combining optional `?`, parentheses, and `null` inside the parentheses. Is this correct? And if so, can we simplify it somehow? 

If `omitempty` is set `null` gets converted to `undefined` and it'd simplify to not being the parentheses or the `null`, is that implemented?

Reply:
From my understanding on the `?` optional field, it only according to my reading of the spec means `undefined` - `null` is different to undefined. -> https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#optional-properties

`omitempty` isn't possible on the AuthParams. so it will neer be undefined here.